include/onnxruntime/core/framework/data_types.h

```diff
@@ -936,7 +918,7 @@ class PrimitiveDataTypeBase : public DataTypeImpl {
   }
 
  protected:
-  PrimitiveDataTypeBase() = default;
+  PrimitiveDataTypeBase(size_t size) : DataTypeImpl{GeneralType::kNonTensor, size}, data_type_{-1} {}
```

Comment:
what does the data_type_ value of -1 mean?

Reply:
Will update and remove SetDataType.